types/assets.go

```diff
@@ -0,0 +1,313 @@
+package types
+
+import (
+	"encoding/binary"
+	"github.com/pkg/errors"
+	"io"
+)
+
+const (
+	AssetsDepth = 24
+)
+
+type AssetInfo struct {
+	Tick          uint32
+	UniverseIndex uint32
+	Siblings      [AssetsDepth][32]byte
+}
+
+func (ai *AssetInfo) UnmarshallBinary(r io.Reader) error {
+	err := binary.Read(r, binary.LittleEndian, &ai.Tick)
+	if err != nil {
+		return errors.Wrap(err, "reading asset tick")
+	}
+
+	err = binary.Read(r, binary.LittleEndian, &ai.UniverseIndex)
+	if err != nil {
+		return errors.Wrap(err, "reading asset universe index")
+	}
+
+	/* We don't read siblings as it doesn't seem to be implemented on node side yet. */
```

Comment:
we need to make sure that we read everything that the node is sending as those connections will be reused, so are you sure that the node is not sending the siblings?